### PR TITLE
SampleGen: write a string or bytes field of the response object to a local file

### PR DESCRIPTION
**Pull request by hzyi-google**
```yaml
- on_success
    - write_file
        contents: $resp.i.am.a.bytes.field
        file_name:
        - "format_%s.mp3"
        - $resp.replace.the.placeholder.with.me
```

### CODE REVIEW

**Comment by hzyi-google** [comment]
cc/ @googleapis/samplegen

**Comment by hzyi-google** [comment]
PTAL

File: cmd/gen-go-sample/testdata/sample_write_file.want [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	req := &foopb.InputType{
+	}
+	resp, err := c.UnaryMethod(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	if err := ioutil.WriteFile(my_bob.mp3, resp.GetDataBob(), 0644); err != nil {
```
Comment (noahdietz) [inline cmd/gen-go-sample/testdata/sample_write_file.want]: I think `my_bob.mp3` should have double quotes: `"my_bob.mp3"`

File: cmd/gen-go-sample/out.go [hunk 2]
```diff
@@ -133,24 +139,50 @@ func writeDefine(txt string, st *symTab, gen *generator) error {
...
+	}
+
+	fn := fnFmt
+	if len(argList) != 0 {
+		fn = fmt.Sprintf("fmt.Sprintf(%q%s)", fmtStrReplacer.Replace(fnFmt), argList)
+	}
+
+	sc, report := initScanner(contPath)
+	cont, typ, err := writePath(sc, st, gen.descInfo)
+
```
Comment (noahdietz) [inline cmd/gen-go-sample/out.go]: can you delete the line break here, bringing the error check next to its source.

File: cmd/gen-go-sample/testdata/sample_write_file.want (hunk 1 above)
Comment (hzyi-google) [reply cmd/gen-go-sample/testdata/sample_write_file.want]: Done!

File: cmd/gen-go-sample/out.go (hunk 2 above)
Comment (hzyi-google) [reply cmd/gen-go-sample/out.go]: Done!

File: cmd/gen-go-sample/testdata/sample_write_file.want [hunk 3]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2018 Google LLC
```
Comment (vchudnov-g) [inline cmd/gen-go-sample/testdata/sample_write_file.want]: 2019

**Pull request merged**